recorder/src/test/groovy/com/focusit/recorder/specs/InternalRecorderSpec.groovy

```diff
@@ -0,0 +1,47 @@
+import com.focusit.jsflight.recorder.internalevent.InternalEventRecorderBuilder
+import org.awaitility.Duration
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.util.concurrent.Callable
+
+import static org.awaitility.Awaitility.await
+
+class InternalRecorderSpec extends Specification {
+
+    @Shared
+    File file = new File("/tmp/record/test")
+
+    def setup() {
+        file.parentFile.mkdirs()
+    }
+
+    def "Records with timestamp after rollover date should be stored in a new file"() {
+        File file = new File("/tmp/record/test")
+        file.parentFile.mkdirs()
+        given:
+        def recorder = InternalEventRecorderBuilder.builderFor(file.getAbsolutePath()).rolloverStrategy(1000).build();
+        recorder.openFileForWriting()
+        recorder.startRecording()
+        await('Single file should be created').until({
+            def files = file.getParentFile().listFiles()
+            return files?.size() == 1 && files?.find { f -> f.name.startsWith('test') }
+        } as Callable<Boolean>)
+        def fileName = file.getParentFile().listFiles().first().getName();
+        when:
+        await().pollDelay(Duration.TWO_SECONDS).until({
+            recorder.push("test", new Object())
+        } as Runnable);
+
+        then:
+        await().until({
```

Comment:
Just for a note.

then block should have some checks or should explicitly return boolean value. 
There is no explicit check/return. there's just wait until something. looks awkward.